### PR TITLE
Write documentation for sugar3.graphics.window

### PR DESCRIPTION
**Pull request by mjuvekar7**
Write documentation for the sugar3.graphics.window module

### CODE REVIEW

File: src/sugar3/graphics/window.py [hunk 1]
```diff
@@ -131,11 +145,11 @@ def __init__(self, **args):
...
     def reveal(self):
-        """ Make window active
+        """
+        Make window active.
 
-        In contrast with present(), brings window to the top
-        even after invoking on response on non-gtk events.
-        See #1423.
+        Brings the window to the top and makes it acive, even after invoking on
+        response to non-gtk events (in contrast to present()).
```
Comment (samdroid-apps) [inline src/sugar3/graphics/window.py]: Keeping the bug number would be optimal

File: src/sugar3/graphics/window.py [hunk 2]
```diff
@@ -84,6 +90,14 @@ def _screen_size_changed_cb(self, screen):
 
 
 class Window(Gtk.Window):
+    """
+    UI interface for activity Windows
+
+    Windows are used as a container to display things that happen in an
+    activity. They contain canvas content, alerts messages, a tray and a
+    toolbar. Windows can either be in fullscreen or windowed (unfullscreen)
+    mode.
```
Comment (samdroid-apps) [inline src/sugar3/graphics/window.py]: Please explain how full screen relates to the visibility of the toolbar.

File: src/sugar3/graphics/window.py [hunk 3]
```diff
@@ -147,9 +161,18 @@ def reveal(self):
...
+        Check whether the window is fullscreen or not.
+
+        Returns:
+            bool: true if window is fullscreen, false otherwise
+        """
         return self._is_fullscreen
 
     def fullscreen(self):
+        """
+        Make the window fullscreen.
```
Comment (samdroid-apps) [inline src/sugar3/graphics/window.py]: Hide the toolbar

File: src/sugar3/graphics/window.py [hunk 4]
```diff
@@ -171,6 +194,9 @@ def fullscreen(self):
                     self.__unfullscreen_button_timeout_cb)
 
     def unfullscreen(self):
+        """
+        Put the window in windowed mode (not fullscreen).
```
Comment (samdroid-apps) [inline src/sugar3/graphics/window.py]: Shows the toolbar

File: src/sugar3/graphics/window.py [hunk 5]
```diff
@@ -186,6 +212,12 @@ def unfullscreen(self):
                 self._unfullscreen_button_timeout_id = None
 
     def set_canvas(self, canvas):
+        """
+        Set current canvas of the window.
+
+        Args:
+            canvas (:class:`Gtk.DrawingArea`): the canvas to set as current
```
Comment (samdroid-apps) [inline src/sugar3/graphics/window.py]: Nop, just a Gtk.Widget

File: src/sugar3/graphics/window.py [hunk 6]
```diff
@@ -196,14 +228,32 @@ def set_canvas(self, canvas):
         self.__vbox.set_focus_child(self._canvas)
 
     def get_canvas(self):
+        """
+        Get current canvas content of the window.
+
+        Returns:
+            Gtk.DrawingArea: the current canvas of the window
```
Comment (samdroid-apps) [inline src/sugar3/graphics/window.py]: Gtk.Widget

File: src/sugar3/graphics/window.py [hunk 7]
```diff
@@ -196,14 +228,32 @@ def set_canvas(self, canvas):
...
+            sugar3.graphics.toolbar_box: the current toolbar box of the window
+        """
         return self._toolbar_box
 
     def set_toolbar_box(self, toolbar_box):
+        """
+        Set window's current toolbar.
+
+        Args:
+            toolbar_box (:class:`sugar3.graphics.toolbarbox.ToolbarBox`): the toolbar box to set as current
```
Comment (samdroid-apps) [inline src/sugar3/graphics/window.py]: Which type?  You've put a different one for the getter and setter

File: src/sugar3/graphics/window.py [hunk 8]
```diff
@@ -216,6 +266,13 @@ def set_toolbar_box(self, toolbar_box):
     toolbar_box = property(get_toolbar_box, set_toolbar_box)
 
     def set_tray(self, tray, position):
+        """
+        Set the window's current tray.
+
+        Args:
+            tray (:class:`sugar3.graphics.tray`): the tray to set
```
Comment (samdroid-apps) [inline src/sugar3/graphics/window.py]: Try is a module, what class?

File: src/sugar3/graphics/window.py [hunk 9]
```diff
@@ -230,6 +287,12 @@ def set_tray(self, tray, position):
         self.tray = tray
 
     def add_alert(self, alert):
+        """
+        Add an alert message to the window.
+
+        Args:
+            alert (:class:`sugar3.graphics.alert.Alert`): the alert to add
```
Comment (samdroid-apps) [inline src/sugar3/graphics/window.py]: Do I need to .show an alert or is this included?

File: src/sugar3/graphics/window.py [hunk 10]
```diff
@@ -320,9 +389,21 @@ def __size_allocate_cb(self, widget, allocation):
         self.__vbox.size_allocate(allocation)
 
     def set_enable_fullscreen_mode(self, enable_fullscreen_mode):
+        """
+        Set the `_enable_fullscreen_mode` variable.
```
Comment (samdroid-apps) [inline src/sugar3/graphics/window.py]: Don't tell me  abou _enable_fullscreen_mode - just tell me what it is.

**Comment by mjuvekar7** [comment]
I have fixed the problems that you commented about. Please tell me if there is something else to change (my GCI task is at https://codein.withgoogle.com/dashboard/task-instances/5747432323284992/)

**Comment by samdroid-apps** [comment]
Please submit ONE fixed commit.

Also, maybe "windowed" mode should be non-fullscreen as sugar doesn't have windows in the traditional sense?

Thanks for working on the task.

**Comment by mjuvekar7** [comment]
Done.

**Comment by samdroid-apps** [comment]
Looks great.  I'm afk for a few days, so anybody else is welcome to do the merge (@godiard or @tchx84)?

**Comment by samdroid-apps** [comment]
Looks great.  Eventually this will be up on a website for reference!

**Pull request merged**